addressing/README.md

```diff
@@ -0,0 +1,334 @@
+peers exchange addresses over the wire in the [identify protocol][identify-spec]
+and other core libp2p protocols.
+
+#### Historical Note: the `ipfs` multiaddr Protocol
+
+The `p2p` multiaddr protocol was originally named `ipfs`, and we've been eliminating
+support for the ipfs string representation of this multiaddr component. It may be
+printed as `/ipfs/<peer-id>` instead of `/p2p/<peer-id>` in its string representation
+depending on the implementation in use. Both names resolve to the same protocol code,
+and they are equivalent in the binary form.
+
+
+## Transport multiaddrs
+
+Because multiaddr is an open and extensible format, it's not possible to
+guarantee that any valid multiaddr is semantically meaningful or usable in a
+particular network. For example, the `/tcp/42` multiaddr, while valid, is not
+useful on its own as a locator.
+
+This section covers the types of multiaddr supported by libp2p transports. It's
+possible that this section will go out of date as new transport modules are
+developed, at which point pull-requests to update this document will be greatly
+appreciated.
+
+### IP and Name Resolution
+
+Most libp2p transports use the IP protocol as a foundational layer, and as a
+result, most transport multiaddrs will begin with a component that represents an
+IPv4 or IPv6 address. 
+
+This may be an actual address, such as `/ip4/7.7.7.7` or
+`/ip6/fe80::883:a581:fff1:833`, or it could be something that resolves to an IP
+address, like a domain name.
+
+libp2p will attempt to resolve "name-based" addresses into IP addresses. The
+current [multiaddr protocol table][multiaddr-proto-table] defines four
+resolvable or "name-based" protocols:
+
+| protocol  | description                                                        |
+|-----------|--------------------------------------------------------------------|
+| `dns`     | Resolves DNS A and AAAA records into both IPv4 and IPv6 addresses. |
+| `dns4`    | Resolves DNS A records into IPv4 addresses.                        |
+| `dns6`    | Resolves DNS AAAA records into IPv6 addresses.                     |
+| `dnsaddr` | Resolves multiaddrs from a special TXT record.                     |
+
+
+When the `/dns` protocol is used, the lookup may result in both IPv4 and IPv6
+addresses, in which case IPv6 will be preferred. To explicitly resolve to IPv4
+or IPv6 addresses, use the `/dns4` or `/dns6` protocols, respectively. 
+
+Note that in some restricted environments, such as inside a web browser, libp2p
+may not have access to the resolved IP addresses at all, in which case the
+runtime will determine what IP version is used.
+
+When a name-based multiaddr encapsulates another multiaddr, only the name-based
+component is affected by the lookup process. For example, if `example.com`
+resolves to `1.2.3.4`, libp2p will resolve the address
+`/dns4/example.com/tcp/42` to `/ip4/1.2.3.4/tcp/42`.
+
+#### dnsaddr Links
```

Comment:
A couple of things here:
1. Should we reference the official dnslink spec: https://dnslink.io/
2. Should we make clear that we use dnsaddrs vs. the spec uses dnslink?  I assume this is intentional.
Don't block, but mentioning @lidel given his work on cleaning this stuff up on the dnslink side.

Reply:
dnsaddr and dnslink are mostly unrelated (except that they both use TXT records). We _should_ probably reference dnslink just to make it clear that they're not the same thing.

Reply:
I added a short note describing how they're different. That should be sufficient to avoid people confusing the two.